CMakeLists.txt

```diff
@@ -553,6 +555,32 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
 
         SET(CMAKE_C_FLAGS_RELEASE     "/GL /O1b2s")
         SET(CMAKE_C_FLAGS_DEBUG       "/Od")
+
+    elseif(TOOLCHAIN STREQUAL "ARM_CLANG")
+        set(TOOLCHAIN_PATH C:/ArmStudio/sw/ARMCompiler6.18)
+        find_program(ARM_CLANG NAMES armclang PATHS ${TOOLCHAIN_PATH}/bin REQUIRED NO_DEFAULT_PATH)
+        find_program(ARM_CXX NAMES armclang PATHS ${TOOLCHAIN_PATH}/bin REQUIRED NO_DEFAULT_PATH)
+        find_program(ARM_ASM NAMES armasm PATHS ${TOOLCHAIN_PATH}/bin REQUIRED NO_DEFAULT_PATH)
+        find_program(ARM_AR NAMES armar PATHS ${TOOLCHAIN_PATH}/bin REQUIRED NO_DEFAULT_PATH)
+        find_program(ARM_LINK NAMES armlink PATHS ${TOOLCHAIN_PATH}/bin REQUIRED NO_DEFAULT_PATH)
+        find_program(ARM_FROMELF NAMES fromelf PATHS ${TOOLCHAIN_PATH}/bin REQUIRED NO_DEFAULT_PATH)
+
+        set(CMAKE_C_COMPILER ${ARM_CLANG} CACHE STRING "")
+        set(CMAKE_CXX_COMPILER ${ARM_CXX} CACHE STRING "")
+        set(CMAKE_ASM_COMPILER ${ARM_ASM} CACHE STRING "")
+        set(CMAKE_LINKER ${ARM_LINK} CACHE STRING "")
+
+        set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+        set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+        set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+        set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+        set(CMAKE_FIND_ROOT_PATH ${TOOLCHAIN_PATH})
+
+        ADD_COMPILE_OPTIONS(--target=arm-arm-none-eabi)
```

Comment:
This is 32bit ARM.
I think we also need support AARCH64, which will use "--target=aarch64-arm-none-eabi"